api/app/Providers/GraphQLServiceProvider.php

```diff
@@ -139,642 +33,62 @@ static function (): EnumType {
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'AssessmentStepType',
-                    'values' => array_column(AssessmentStepType::cases(), 'name'),
-                ]);
-            }
-        );
-        $typeRegistry->registerLazy(
-            'AssessmentResultType',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'AssessmentResultType',
-                    'values' => array_column(AssessmentResultType::cases(), 'name'),
-                ]);
-            }
-        );
-        $typeRegistry->registerLazy(
-            'AssessmentDecision',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'AssessmentDecision',
-                    'values' => array_column(AssessmentDecision::cases(), 'name'),
-                ]);
-            }
-        );
-        $typeRegistry->registerLazy(
-            'AssessmentResultJustification',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'AssessmentResultJustification',
-                    'values' => array_column(AssessmentResultJustification::cases(), 'name'),
-                ]);
-            }
-        );
-        $typeRegistry->registerLazy(
-            'AssessmentDecisionLevel',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'AssessmentDecisionLevel',
-                    'values' => array_column(AssessmentDecisionLevel::cases(), 'name'),
-                ]);
-            }
-        );
-        $typeRegistry->registerLazy(
-            'NotificationFamily',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'NotificationFamily',
-                    'values' => array_column(NotificationFamily::cases(), 'name'),
-                ]);
-            }
-        );
+        $enums = Discover::in(app_path('Enums'))
+            ->enums()
+                // Language has a custom implementation
+            ->custom(fn (DiscoveredStructure $structure) => $structure->name !== 'Language')
+            ->sortBy(Sort::Name)
+            ->get();
 
-        $typeRegistry->registerLazy(
-            'CandidateRemovalReason',
-            static function (): EnumType {
-                return new EnumType([
-                    'name' => 'CandidateRemovalReason',
-                    'values' => array_column(CandidateRemovalReason::cases(), 'name'),
-                ]);
-            }
-        );
+        foreach ($enums as $enum) {
+            $name = class_basename($enum);
+            $typeRegistry->registerLazy(
+                $name,
+                static function () use ($name, $enum): EnumType {
+                    /** @disregard P1013 */
+                    $values = array_column($enum::cases(), 'name');
```

Comment:
Does anyone know how to typehint this so we don't need to disregard the error about missing `cases()` method? :thinking:

Reply:
I think this would do it?
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/2ebc6e7e-200e-4964-ac5d-fbc9aae6fb0b)

Reply:
Of course it was `UnitEnum` I was trying just `Enum` :woman_facepalming:

Reply:
Yeah, it's a bit of a weird one.  Took some googling until I found https://www.php.net/manual/en/class.unitenum.php